OPHD/Map/TileMap.cpp

```diff
@@ -608,10 +608,33 @@ void TileMap::AdjacentCost(void* state, std::vector<micropather::StateCost>* adj
 		Tile* adjacent_tile = getTile(tilePosition + offset, 0);
 		float cost = constants::ROUTE_BASE_COST;
 
-		if (!adjacent_tile || !adjacent_tile->empty() || adjacent_tile->index() == TerrainType::TERRAIN_IMPASSABLE) { cost = FLT_MAX; }
-		else { cost *= static_cast<float>(adjacent_tile->index()) + 1.0f; }
+		if (!adjacent_tile || adjacent_tile->index() == TerrainType::TERRAIN_IMPASSABLE)
+		{
+			cost = FLT_MAX;
+		}
+		else if (adjacent_tile && !adjacent_tile->empty())
+		{
+			if (adjacent_tile == mPathStartEndPair.first || adjacent_tile == mPathStartEndPair.second)
+			{
+				cost *= static_cast<float>(adjacent_tile->index()) + 1.0f;
```

Comment:
It's not super clear to me at first glance why `cost` is being multiplied, particularly in an accumulator style way. It would make more sense to me if this was written here as:
```cpp
cost = constants::ROUTE_BASE_COST * static_cast<float>(adjacent_tile->index()) + 1.0f;
```

The code becomes easier to understand when you don't have to go hunting down a previous assignment.

Reply:
I would agree, as the comment notes, this can be done better but for whatever reason I'm just not seeing it.